### ISSUE TITLE
[Feature Request] Add a Status column in Emulator GUI

### ISSUE THREAD

**Issue by Crispy81** [shadps4-emu/shadPS4#1571]
Be good to see the reported status of any installed games via the Emulator gui ie - Playable, Ingame, Menus etc...

Maybe a reference to the tags of reported games? It would save time having to look up the status of games you have installed, especially if it's a long list. If there is no report for a game you have installed, then it could be something as generic as 'No status found' or something.

**Comment by Hermiten** [comment]
This can be a huge addition to the launcher !
We have to find a dev available for this 👍

**Comment by f8ith** [comment]
~~Scraping Github issues requires an API key, so to implement this we probably need to host another file somewhere with game compatibility information~~

Nevermind the API works without a key, I'll see if I can get something working

**Comment by GHU7924** [comment]
Yes, something like this would be very useful.

You can do something similar to checking compatibility for Steam Deck.
Four categories of Deck compatibility: Verified, Playable, Unsupported, Unknown.

For shadPS4: Playable (Green), Ingame (Yellow), Intro/Menus/Loadable (Red), Nothing (Gray).

There is also a website shadps4.net (but perhaps this is a more difficult path.)

**Comment by georgemoralis** [comment]
I have already wrote a script to fetch compatibility list repo to a database for creating a compatibility list for the site . Probably i will write an API that return a JSON so it can update shadps4 UI too @DanielSvoboda what you think about this?

**Comment by GHU7924** [comment]
In addition, there is something else that I would like to suggest, so I will combine everything in one drawing.
The Windows and Linux versions of shadPS4 are different, but it would be great if they looked similar.
I'm talking about Log now. In Windows we see 3 windows, in Linux - 2.
On Windows, Log scares beginners, especially if there are a lot of red lines, and on Linux you don't see what's going on at all until you go to the Log folder.
If it is possible to implement this, then I offer you my vision of the shadPS4 window. A status column has also been added.
It is also worth making a description of the statuses somewhere, how they differ and what is peculiar to each of them.
![878](https://github.com/user-attachments/assets/f7a02308-043d-4725-95cc-eb3657a9f0f9)

**Comment by DanielSvoboda** [comment]
> I have already wrote a script to fetch compatibility list repo to a database for creating a compatibility list for the site . Probably i will write an API that return a JSON so it can update shadps4 UI too @DanielSvoboda what you think about this?

It seems like a good idea, you just need to understand how it will all work, for example. whenever you open the emulator it will make a request, or have a button similar to rpcs3 'Download Compatibility Database'. It will be stored in a single file or a file for each game. It will be downloaded from all games or individually to the one you already have installed. These and other points have to be thought about before starting to do it, but it's a good idea

**Comment by GHU7924** [comment]
@DanielSvoboda You're right, it really needs to be thought through well. 
I'll write a couple of my thoughts. I don't know if it will be possible to implement this, but at least I will suggest it.

>whenever you open the emulator it will make a request, or have a button similar to rpcs3 'Download Compatibility Database'.

I think we can combine this process with updating the emulator. A window appears in front of you saying that there is a new version of the emulator, you click update, the download begins and at the same time the compatibility database is checked, if it has also been updated, then it is loaded.
About the database download button... I don't know... Maybe YES, maybe NOT.
I can say the same about Check for Updates in the Help section, because in the settings on the General tab we can also check for updates.

>It will be stored in a single file or a file for each game. It will be downloaded from all games or individually to the one you already have installed.

Since this is a database, I think that all the games should be in one file.
As in the case of patches, the entire database should be downloaded, which adapts to you.
Here you have found a new game for yourself, added it to the emulator and you can immediately see the status of the game, because you have already downloaded the database.

The only point is that the database should not be one, there should be several, maybe 8... pieces.
I explain why this is so. We have several operating systems: Windows, Linux, macOS and Ubuntu (since I don't understand much about Linux, I don't know why it goes separately, but let's say).
What does it mean?.. Based on this information, we should have 4 databases, that is, one file for each operating system.
It seems that, for example, Bloodborne works well on Windows, but is not supported on macOS (if I understood correctly) or Beach Buggy Racing 2 I can already play on Windows, but I could not run it on Steam Deck OLED.
That is, it turns out that the performance and the ability to run games on different operating systems may differ.

So why did I name the number 8 and not 4 or 1 for the number of databases... It's just because
we also have Nightly and Release channels. What didn't work in the Release version can work in Nightly.
Again, for example, Beach Buggy Racing 2 did not work in v0.4.0, but it works now in v0.4.1 (after improvements to saves).
We multiply 4 databases by 2 update channels and get 8 databases.

I don't know about the names, you can make abbreviations, for example:

shadPS4 Game Compatibility Database Windows Nightly = shad_GCD_Win_N
shadPS4 Game Compatibility Database Windows Release = shad_GCD_Win_R

shadPS4 Game Compatibility Database Linux Nightly = shad_GCD_Lin_N
shadPS4 Game Compatibility Database Linux Release = shad_GCD_Lin_R

To summarize, depending on the operating system and the update channel, a certain database must be downloaded.
While there is no separation and other things, you can specify that a specific database is relevant for the emulator version v0.4.0 (or v0.4.1), in test mode, so to speak. Subsequently, version 0.4.1 will become version 0.5.0, and then 0.5.1 will become 0.6.0

**Comment by f8ith** [comment]
> To summarize, depending on the operating system and the update channel,

iirc, I don't think the game compatibility database allows testing on nightly. 

> Ubuntu (since I don't understand much about Linux, I don't know why it goes separately, but let's say).

Ubuntu and Linux are the same thing, we just have 3 platforms

I believe it'd be easier to have a master file containing the whole database, and just have emulator select the best "status" for the users' current platform for any given game. Some games only have an issue for windows, some have windows + linux etc.

**Comment by GHU7924** [comment]
>I don't think the game compatibility database allows testing on nightly.

I'm not saying that testing needs to be done for every Nightly build.
You can do test runs once a week and update information based on them.

>Ubuntu and Linux are the same thing

I know that, but you have the option to download Linux-Qt, Linux-SDL and Ubuntu64 builds, that's why I mentioned it.
This may be due to a different file format, but yes, we have 3 platforms, then there will be 6 databases, which simplifies the task to some extent.

>I believe it'd be easier to have a master file containing the whole database, and just have emulator select the best "status" for the users' current platform for any given game.

If it is possible that one file contains all databases, and the emulator extracts the necessary information according to the platform and release on which it is currently running, then that would be great. In addition, we cannot afford to choose the best status, since the emulator is in alpha version, and in case of improvements and functional changes in some games, both progress and regression may occur. The previous build started the game, but the new one may somehow affect it and add problems, or it may stop running.

>Some games only have an issue for windows, some have windows + linux etc.

That's why I'm saying that we need several databases.

**Comment by Hermiten** [comment]
Implemented #1668
